internal/handlers/common/aggregations/expression.go

```diff
@@ -97,54 +92,54 @@ func NewExpressionWithOpts(expression string, opts *ExpressionOpts) (Expression,
 		// `$$` indicates field is a variable.
 		v := strings.TrimPrefix(expression, "$$")
 		if v == "" {
-			return nil, newFieldPathError(ErrEmptyVariable)
+			return nil, newExpressionError(ErrEmptyVariable)
 		}
 
 		if strings.HasPrefix(v, "$") {
-			return nil, newFieldPathError(ErrInvalidFieldPath)
+			return nil, newExpressionError(ErrInvalidExpression)
 		}
 
 		// TODO https://github.com/FerretDB/FerretDB/issues/2275
-		return nil, newFieldPathError(ErrUndefinedVariable)
+		return nil, newExpressionError(ErrUndefinedVariable)
 	case strings.HasPrefix(expression, "$"):
 		// `$` indicates field is a path.
 		val = strings.TrimPrefix(expression, "$")
 
 		if val == "" {
-			return nil, newFieldPathError(ErrEmptyFieldPath)
+			return nil, newExpressionError(ErrEmptyFieldPath)
 		}
 	default:
-		return nil, newFieldPathError(ErrNotFieldPath)
+		return nil, newExpressionError(ErrNotExpression)
 	}
 
 	var err error
 
-	path, err := NewPathFromString(val)
+	path, err := types.NewPathFromString(val)
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
-	return &pathExpression{
+	return &Expression{
 		path:           path,
 		ExpressionOpts: opts,
 	}, nil
 }
 
 // NewExpression creates a new instance by checking expression string.
-func NewExpression(expression string) (Expression, error) {
+func NewExpression(expression string) (*Expression, error) {
 	// TODO https://github.com/FerretDB/FerretDB/issues/2348
 	return NewExpressionWithOpts(expression, new(ExpressionOpts))
 }
 
 // Evaluate gets the value at the path.
-func (p *pathExpression) Evaluate(doc *Document) any {
-	path := p.path
+func (e *Expression) Evaluate(doc *types.Document) any {
+	path := e.path
 
 	if path.Len() == 1 {
 		val, err := doc.Get(path.String())
 		if err != nil {
 			// if the path does not exist, return nil.
-			return Null
+			return types.Null
```

Comment:
Could you replace the comment with why we return Null please?

Reply:
done